spec/models/course_spec.rb

```diff
@@ -434,4 +434,103 @@
       its(:content_status) { should eq(:published_with_unpublished_changes) }
     end
   end
+
+  describe "#publish_enrichments" do
```

Comment:
I think this should live in its own file `course_publish_spec.rb`

Reply:
I like the pattern where the contents of this describe block would go into:

```
app/models/course/publish_enrichments_spec.rb
```

Naming it after the method being focused on makes it super easy to find. Thoughts? Is it too granular?

Reply:
I'm not a massive fan of this. I would find it confusing if the spec file matching the course model file doesn't have all the tests.

This is a side effect of course becoming a bit of a god object, we should look at decomposing it. One way to do that would be to move some code into a concern, then testing that in its own file.

I think this would be too granular and ultimately confusing.

Reply:
Given this requires further discussion I suggest we merge this as-is then make improvements as future PRs.